src/url.cpp

```diff
@@ -22,8 +22,34 @@ namespace Url
     const std::string Url::QUERY = Url::PCHAR + "/?";
     const std::string Url::FRAGMENT = Url::PCHAR + "/?";
     const std::string Url::USERINFO = Url::UNRESERVED + Url::SUB_DELIMS + ":";
+    const std::string Url::HEX = "0123456789ABCDEF";
     const std::string Url::SCHEME =
         "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789+-.";
+    const std::vector<int> Url::HEX_TO_DEC = {
```

Comment:
I'd recommend you use `std::vector<signed char>` for some space savings. You're currently using 64-bits per entry when you don't need more than 8-bits.